packages/components/src/public/QueryModel/ChartMenu.tsx

```diff
@@ -1,16 +1,37 @@
-import React, { PureComponent, ReactNode } from 'react';
-import { DropdownButton, MenuItem } from 'react-bootstrap';
+import React, { FC, PureComponent, ReactNode, SyntheticEvent, useCallback } from 'react';
+import { DropdownButton, MenuItem, SelectCallback } from 'react-bootstrap';
 
-import { ChartMenuItem } from '../../internal/components/chart/ChartMenuItem';
 import { DataViewInfo } from '../../internal/DataViewInfo';
+import { cancelEvent } from '../../internal/events';
 
-import { ChartModal } from '../../internal/components/chart/ChartModal';
 import { blurActiveElement } from '../../internal/util/utils';
 
 import { getQueryMetadata } from '../../internal/global';
 
 import { RequiresModelAndActions } from './withQueryModels';
 
+interface ChartMenuItemProps {
```

Comment:
Interaction feedback. When opening a report the chart menu stays open. Is there a way to have this close after selection?

<img width="1466" alt="image" src="https://github.com/LabKey/labkey-ui-components/assets/3926239/c9e3fe1d-2bce-4b1f-be4e-cc17478d1681">

Reply:
Not that I could find, but I'll try again.

Reply:
This is an issue with every menu in the GridPanel ButtonBar, it seems to be an issue with React Bootstrap. This doesn't happen on our grid column menus because we don't use the DropdownButton or DropdownMenu components, we render our own manually, and directly compute the visibility and positioning.

We should probably roll our own version of DropdownMenu and DropdownButton that doesn't have this issue, and will let us render components that render MenuItems, so we can make more re-usable components. We need to do this anyway in order to move away from react-bootstrap, as part of clearing the path to upgrade React.

Reply:
https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=48576